application/classes/Ushahidi/Repository/Form/Contact.php

```diff
@@ -81,7 +83,17 @@ public function updateCollection(Array $entities, $form_id = null)
 
 		// Start transaction
 		$this->db->begin();
+		$invalidatedContacts =  [];
 		foreach ($entities as $entity) {
+			$contactOnActiveSurvey = $this->existsInActiveTargetedSurveyByContactNumber($entity->contact);
+			if ($contactOnActiveSurvey) {
+				$this->setInactiveTargetedSurvey($contactOnActiveSurvey['targeted_survey_state_id']);
+				$invalidatedContacts[] = [
+					'contact' => $contactOnActiveSurvey['contact'],
+					'contact_id' => $contactOnActiveSurvey['contact_id'],
+					'form_id' => $contactOnActiveSurvey['form_id']
+				];
+			}
 			//@fixme how to avoid this ugly line?
```

Comment:
If we're going to live with this ugly line for now, can we at least explain why it's here?

Reply:
good point. updating